tastebuddies/__init__.py

```diff
@@ -0,0 +1,76 @@
+from pyramid.config import Configurator
+from sqlalchemy import engine_from_config
+
+import os
+
+from pyramid.authentication import AuthTktAuthenticationPolicy
+from pyramid.authorization import ACLAuthorizationPolicy
+from cryptacular.bcrypt import BCRYPTPasswordManager
+
+from .models import (
+    DBSession,
+    Base,
+)
+
+from security import groupfinder, Root, UserFactory
+from pyramid.view import forbidden_view_config
+
+
+def main(global_config, **settings):
+    """ This function returns a Pyramid WSGI application.
+    """
+
+    @forbidden_view_config(renderer='templates/forbidden.jinja2')
+    def forbidden(request):
+        return {}
+
+    engine = engine_from_config(settings, 'sqlalchemy.')
+    DBSession.configure(bind=engine)
+    Base.metadata.bind = engine
+
+    auth_secret = os.environ.get('TASTEBUDDIES_AUTH_SECRET', 'tastynoms')
+
+    authn_policy = AuthTktAuthenticationPolicy(
+        secret=auth_secret,
+        hashalg='sha512',
+        callback=groupfinder,
+    )
+
+    authz_policy = ACLAuthorizationPolicy()
+
+    settings['auth.username'] = os.environ.get('AUTH_USERNAME', 'admin')
+    manager = BCRYPTPasswordManager()
+    settings['auth.password'] = os.environ.get(
+        'AUTH_PASSWORD', manager.encode('secret')
+    )
```

Comment:
now that you have a system with users, you should get rid of this configuration hack.